api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -109,6 +109,8 @@ public Schema(int schemaId, List<NestedField> columns, Map<String, Integer> alia
   static void validateIdentifierField(int fieldId, Map<Integer, Types.NestedField> idToField,
                                               Map<Integer, Integer> idToParent) {
     Types.NestedField field = idToField.get(fieldId);
+    Preconditions.checkArgument(field != null,
+        "Can not add filedId %d as an identifier field: field not exists", fieldId);
```

Comment:
Typo 'fileId'.  Should we put something like 'field with id %s'?

Also, I thought Preconditions only supports %s, does this actually work?  
https://javadoc.io/doc/com.google.guava/guava/latest/com/google/common/base/Preconditions.html

Reply:
> Also, I thought Preconditions only supports %s, does this actually work?
https://javadoc.io/doc/com.google.guava/guava/latest/com/google/common/base/Preconditions.html

You are right. I changed to %s now.

Reply:
> maybe change it to Preconditions.checkArgument(null != field, "Cannot use fieldId %s: field does not exist", fieldId);

I replaced `can not` with `cannot` which is also suggested in another comment. I think `filed != null` reads smoother and I see more places use this pattern. Keep "as an identifier field" in error message as this error may happen when constructing a full Schema, where it's clearer to report the error is for identifier fields.